lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -83,9 +84,11 @@ export class AuthCache extends BrowserStorage {// Singleton
      * @param value
      * @param enableCookieStorage
      */
-    setItem(key: string, value: string, enableCookieStorage?: boolean, state?: string): void {
+    setItem(key: string, value: string, enableCookieStorage?: boolean): void {
         super.setItem(this.generateCacheKey(key, true), value, enableCookieStorage);
-        if (this.rollbackEnabled) {
+
+        // Values stored in cookies will have rollback disabled to minimize cookie length
+        if (this.rollbackEnabled && !enableCookieStorage) {
             super.setItem(this.generateCacheKey(key, false), value, enableCookieStorage);
         }
```

Comment:
Rollback was generating two cookies for each value, greatly increasing the overall cookie length.

Reply:
What is rollback used for?

Reply:
Storage keys that did not include the client id.